service/access/server.go

```diff
@@ -448,20 +452,23 @@ func (s *Server) ExecuteScriptAtBlockHeight(_ context.Context, in *access.Execut
 	for _, arg := range in.Arguments {
 		val, err := json.Decode(nil, arg)
 		if err != nil {
-			return nil, fmt.Errorf("could not decode script argument: %w", err)
+			err = fmt.Errorf("could not decode script argument: %w", err)
+			return nil, status.Error(codes.Internal, err.Error())
```

Comment:
The user provided arguments were invalid
```suggestion
			return nil, status.Error(codes.InvalidArgument, err.Error())
```